domains/emissions/Emissions.API/API/Controllers/EmissionsController.cs

```diff
@@ -32,15 +22,17 @@ public async Task<ActionResult<EmissionsResponse>> GetEmissions([FromQuery] Ener
             return ValidationProblem(ModelState);
         }
 
-        var dateFromDateTime = request.DateFrom.ToDateTime();
-        var dateToDateTime = request.DateTo.ToDateTime();
-
-        return Ok(await emissionsService.GetTotalEmissions(Context, dateFromDateTime, dateToDateTime, request.Aggregation));
+        return Ok(await emissionsService.GetTotalEmissions(
+            Context,
+            DateTimeOffset.FromUnixTimeSeconds(request.DateFrom),
+            DateTimeOffset.FromUnixTimeSeconds(request.DateTo),
+            request.TimeZoneInfo,
+            request.Aggregation));
     }
 
     [HttpGet]
     [Route("api/sources")]
-    public async Task<ActionResult<EnergySourceResponse>> GetEnergySources([FromQuery] EnergySourceRequest request)
+    public async Task<ActionResult<EnergySourceResponse>> GetEnergySources([FromQuery] EnergySourceRequest request, IEmissionsService emissionsService, IValidator<EnergySourceRequest> validator)
```

Comment:
AFAIK we cannot change the emission service scope, because it needs the two services that gets a HTTP client injected and those clients are considered transient. How ever we could change the calculators to singleton scope.